sdk/textanalytics/Azure.AI.TextAnalytics/src/TextAnalyticsClient.cs

```diff
@@ -157,8 +158,8 @@ public virtual async Task<Response<DetectedLanguage>> DetectLanguageAsync(string
                 if (languageDetection.Results.Errors.Count > 0)
                 {
                     // only one document, so we can ignore the id and grab the first error message.
-                    var error = Transforms.ConvertToError(languageDetection.Results.Errors.FirstOrDefault().Error);
-                    throw await _clientDiagnostics.CreateRequestFailedExceptionAsync(response, new ResponseError(error.ErrorCode.ToString(), error.Message), CreateAdditionalInformation(error)).ConfigureAwait(false);
+                    var error = languageDetection.Results.Errors[0].Error;
```

Comment:
Although technically correct, in TA we always surface the `InnerError` information instead of the first level `Error` .
See for example:
```
"error": {
        "code": "InvalidRequest",
        "message": "Invalid document in request.",
        "innererror": {
            "code": "InvalidDocument",
            "message": "Request contains duplicated Ids. Make sure each document has a unique Id."
        }
    }
```

The real information is in the `InnerError` property, which is why the transform was looking for it and overwriting the `Error` type.

Reply:
hrmm....   yeah, I overlooked that.   I'll fix that up.